src/cassandra_statement.rs

```diff
@@ -71,14 +71,16 @@ pub enum CassandraStatement {
 impl CassandraStatement {
     /// extract the cassandra statement from an AST tree.
     /// the boolean return value is `true` if there is a parsing error in the statement tree.
-    pub fn from_tree(tree: &Tree, source: &str) -> Vec<(bool, CassandraStatement)> {
+    pub fn from_tree(tree: &Tree, source: &str) -> Vec<(bool, CassandraStatement, usize, usize)> {
```

Comment:
Once a tuple reaches this many elements it becomes difficult to understand what each element is for.
So we should replace it with a struct so that we can name each field.

Something like:
```rust
struct ParseStatement {
   error: bool,
   statement: CassandraStatement,
   bytes_parsed_start: usize,
   bytes_parsed_end: usize,
}
```